libraries/AP_Math/matrix3.cpp

```diff
@@ -228,6 +228,42 @@ T Matrix3<T>::det() const
 }
 
 template <typename T>
+Matrix3<T> Matrix3<T>::inverse(bool& success) const
```

Comment:
I think it would be better to return bool and take the Matrix where to save the result as a reference. Then just make sure it still work with *this == other. It will avoid the 3 copies this implementation has in the worst case.